mlflow/models/signature.py

```diff
@@ -393,9 +400,15 @@ def _infer_signature_from_type_hints(
             f"Invalid output type hint, setting output schema to AnyType. Error: {e}", stacklevel=2
         )
         is_output_type_hint_valid = False
-        output_schema = Schema([ColSpec(type=AnyType())])
+        output_schema = default_output_schema
+    if output_schema is None:
+        _logger.warning(
+            f"Type hint {type_hints.output} can not be used to infer model signature "
+            "output schema, setting output schema to AnyType."
```

Comment:
Same as the comment above, should we defer this to input example?

Reply:
Maybe we don't need any warnings... it will be quite confusing if we emit some warning here, but then use input example to infer model signature. I think I'll document how model signature is inferred for pyfunc models in our documentation instead.

Reply:
Yea agreed, maybe we can keep debug log in case if we get any ES ticket:)

Reply:
After a second thought, I updated the logic to check if we should try infer signature from type hints first, and add warning message correspondingly 😆 I feel like if we don't warn anything yea we'll receive tickets